src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -51,33 +56,60 @@ private void CompilationStartAction(CompilationStartAnalysisContext context)
                 // actually necessary.
                 var viewModel = SerializableNamingStylePreferencesInfo.FromXElement(XElement.Parse(currentValue));
                 var preferencesInfo = viewModel.GetPreferencesInfo();
+
+                var idToCachedResult = new Dictionary<Guid, ConcurrentDictionary<string, string>>();
```

Comment:
To slim it down a bit and avoid excessive internal allocations, I would specify a fixed concurrency level (usually 1 or 2 is enough) by using the constructors that takes two ints (the other int is the initial capacity which you can set to zero)